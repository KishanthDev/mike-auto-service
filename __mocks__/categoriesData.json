[
    {
      "category": "Restaurants",
      "subcategories": [
        {
          "name": "Italian",
          "businesses": [
            {
              "id": 1,
              "businessName": "Pasta Palace",
              "gallery": ["https://example.com/image.jpg"],
              "ratings": 4.5,
              "reviews": [{}, {}],
              "description": "Authentic Italian cuisine",
              "highlights": ["Vegan Options", "Outdoor Seating"],
              "contact": {
                "phone": "123-456-7890",
                "website": "https://pastapalace.com"
              },
              "cta": {
                "getDirections": "https://maps.google.com/pastapalace"
              }
            }
          ]
        },
        {
          "name": "Mexican",
          "businesses": []
        }
      ]
    },
    {
      "category": "Cafes",
      "subcategories": [
        {
          "name": "Coffee Shops",
          "businesses": []
        }
      ]
    }
  ]