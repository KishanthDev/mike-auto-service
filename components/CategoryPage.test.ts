import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "./CategoryPage";
import categoriesData from "../data/detailed_categories_with_subcategories.json";

// Mock the next/link component
jest.mock("next/link", () => {
    const MockedLink = ({ children, href }: { children: React.ReactNode; href: string }) => {
      return React.createElement("a", { href }, children);
    };
    MockedLink.displayName = "MockedLink";
    return MockedLink;
  });
  
jest.mock("@/app/lib/slugify", () => ({
  slugify: jest.fn((str) => str.toLowerCase().replace(/\s+/g, "-")),
}));

jest.mock("./breadcrumb/Breadcrumb", () => {
  return () => React.createElement("div", null, "Mocked Breadcrumb");
});

describe("Home Component", () => {
  it("renders the main layout structure", () => {
    render(React.createElement(Home));

    const mainElement = screen.getByRole("main");
    expect(mainElement).toBeInTheDocument();
    expect(mainElement).toHaveClass(
      "flex-1",
      "p-6",
      "overflow-y-auto",
      "scrollbar-hide",
    );
  });

  it("displays the breadcrumb component", () => {
    render(React.createElement(Home));
    expect(screen.getByText("Mocked Breadcrumb")).toBeInTheDocument();
  });

  it("renders the search input", () => {
    render(React.createElement(Home));

    const searchInput = screen.getByPlaceholderText(
      "Search categories, businesses, or services...",
    );
    expect(searchInput).toBeInTheDocument();
    expect(searchInput).toHaveClass(
      "w-full p-2 border rounded-md shadow-md dark:bg-gray-800 dark:border-gray-600",
    );
  });

  it('displays the "Explore Categories" heading', () => {
    render(React.createElement(Home));

    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Explore Categories");
    expect(heading).toHaveClass("text-2xl", "font-semibold", "mb-6");
  });

  it("renders all categories from the data", () => {
    render(React.createElement(Home));

    categoriesData.forEach((category: any) => {
      expect(screen.getByText(category.category)).toBeInTheDocument();
    });
  });

  it("renders up to MAX_VISIBLE_SUBCATS subcategories per category", () => {
    render(React.createElement(Home));

    categoriesData.forEach((category: any) => {
      const visibleSubcats = category.subcategories.slice(0, 4);
      visibleSubcats.forEach((subcat: any) => {
        expect(screen.getByText(subcat.name)).toBeInTheDocument();
      });
    });
  });

  it("applies correct styling to subcategory tags", () => {
    render(React.createElement(Home));

    const firstSubcategory = categoriesData[0].subcategories[0];
    const subcategoryTag = screen.getByText(firstSubcategory.name);

    expect(subcategoryTag).toHaveClass(
      "text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full",
    );
  });

  it("generates correct links for categories", () => {
    render(React.createElement(Home));

    const firstCategory = categoriesData[0];
    const categoryLink = screen.getByText(firstCategory.category).closest("a");

    expect(categoryLink).toHaveAttribute(
      "href",
      `/subcategory/${firstCategory.category.toLowerCase().replace(/\s+/g, "-")}`,
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(React.createElement(Home));
    expect(asFragment()).toMatchSnapshot();
  });
});
