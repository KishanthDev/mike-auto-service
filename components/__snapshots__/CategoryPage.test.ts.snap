// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-screen bg-gray-100 dark:bg-black text-gray-800 dark:text-gray-100"
  >
    <main
      class="flex-1 p-6 overflow-y-auto scrollbar-hide"
    >
      <div
        class="mb-6"
      >
        <div>
          Mocked Breadcrumb
        </div>
        <div
          class="max-w-md mx-auto"
        >
          <input
            class="w-full p-2 border rounded-md shadow-md dark:bg-gray-800 dark:border-gray-600"
            placeholder="Search categories, businesses, or services..."
            type="text"
          />
        </div>
      </div>
      <h2
        class="text-2xl font-semibold mb-6"
      >
        Explore Categories
      </h2>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6"
      >
        <a
          href="/subcategory/üöó-automotive"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üöó Automotive
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Car Repair
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Car Sales
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Tires
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Auto Parts
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üè•-health-&-medical"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üè• Health & Medical
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Hospitals
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Clinics
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Dentists
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Pharmacies
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +2
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üìö-education"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üìö Education
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Schools (Primary, Secondary)
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Colleges & Universities
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Tuition Services
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Educational Consultants
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üçΩÔ∏è-restaurants-&-food"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üçΩÔ∏è Restaurants & Food
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Fast Food
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Fine Dining
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Caf√©s
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Bakeries
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üõçÔ∏è-shopping-&-retail"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üõçÔ∏è Shopping & Retail
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Clothing Stores
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Electronics Stores
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Jewellery Showrooms
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Grocery Stores
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +3
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üíº-professional-business-services"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üíº Professional Business Services
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Chartered Accountants
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Lawyers
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Marketing Agencies
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Business Consultants
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üè†-home-services"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üè† Home Services
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Interior Designers
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Carpenters
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Plumbers
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Electricians
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +3
            </span>
          </div>
        </a>
        <a
          href="/subcategory/‚úàÔ∏è-travel-&-transportation"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            ‚úàÔ∏è Travel & Transportation
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Airlines
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Hotels
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Car Rentals
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Travel Agencies
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üè¢-real-estate"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üè¢ Real Estate
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Builders and Developers
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Civil Contractors
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Real Estate Agents
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Property Management
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üí≥-financial-services"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üí≥ Financial Services
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Banks
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Loan Agencies
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Insurance Agencies
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Investment Services
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üèõÔ∏è-government-&-community"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üèõÔ∏è Government & Community
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Local Government Offices
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Community Centers
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Libraries
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Post Offices
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üé¨-entertainment-&-media"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üé¨ Entertainment & Media
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Movie Theaters
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Music Stores
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Event Management
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Convention Halls
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üèãÔ∏è-sports-&-recreation"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üèãÔ∏è Sports & Recreation
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Gyms
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Sports Leagues
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Parks
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Recreation Centers
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üêæ-pets-&-animals"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üêæ Pets & Animals
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Pet Stores
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Veterinarians
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Pet Grooming
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Animal Shelters
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üíª-computers-&-mobile"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üíª Computers & Mobile
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Internet Service Providers
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Computer Sales and Repair
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Mobile Sales and Repair
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Web Design
            </span>
            <span
              class="text-xs bg-blue-100 dark:bg-blue-800 text-blue-700 dark:text-blue-300 px-2 py-1 rounded-full"
            >
              +1
            </span>
          </div>
        </a>
        <a
          href="/subcategory/üõ°Ô∏è-safety-&-security"
        >
          <h3
            class="text-lg font-bold mb-2"
          >
            üõ°Ô∏è Safety & Security
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Private Security Services
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Fire Departments
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Police Stations
            </span>
            <span
              class="text-xs bg-gray-200 dark:bg-gray-700 text-gray-800 dark:text-gray-200 px-2 py-1 rounded-full"
            >
              Safety Equipment Suppliers
            </span>
          </div>
        </a>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
